tests/foreman/api/test_permission.py

```diff
@@ -38,28 +38,41 @@ def create_permissions(self, class_target_sat):
         # workaround for setting class variables
         cls = type(self)
         cls.permissions = PERMISSIONS.copy()
-        if class_target_sat.is_upstream:
-            cls.permissions[None].extend(cls.permissions.pop('DiscoveryRule'))
-            cls.permissions[None].remove('app_root')
-            cls.permissions[None].remove('attachments')
-            cls.permissions[None].remove('configuration')
-            cls.permissions[None].remove('logs')
-            cls.permissions[None].remove('view_cases')
-            cls.permissions[None].remove('view_log_viewer')
-
-        result = class_target_sat.execute('rpm -qa | grep rubygem-foreman_openscap')
-        if result.status != 0:
+        if not class_target_sat.is_upstream:
+            cls.permissions.pop('LookupValue')
+
+        rpm_packages = class_target_sat.execute('rpm -qa').stdout
+        if 'rubygem-foreman_rh_cloud' not in rpm_packages:
+            cls.permissions.pop('InsightsHit')
+            cls.permissions[None].remove('generate_foreman_rh_cloud')
+            cls.permissions[None].remove('view_foreman_rh_cloud')
+            cls.permissions[None].remove('dispatch_cloud_requests')
+            cls.permissions[None].remove('control_organization_insights')
+        if 'rubygem-foreman_bootdisk' not in rpm_packages:
+            cls.permissions[None].remove('download_bootdisk')
+        if 'rubygem-foreman_virt_who_configure' not in rpm_packages:
+            cls.permissions.pop('ForemanVirtWhoConfigure::Config')
+        if 'rubygem-foreman_openscap' not in rpm_packages:
             cls.permissions.pop('ForemanOpenscap::Policy')
             cls.permissions.pop('ForemanOpenscap::ScapContent')
             cls.permissions[None].remove('destroy_arf_reports')
             cls.permissions[None].remove('view_arf_reports')
             cls.permissions[None].remove('create_arf_reports')
-        result = class_target_sat.execute('rpm -qa | grep rubygem-foreman_remote_execution')
-        if result.status != 0:
+        if 'rubygem-foreman_remote_execution' not in rpm_packages:
             cls.permissions.pop('JobInvocation')
             cls.permissions.pop('JobTemplate')
             cls.permissions.pop('RemoteExecutionFeature')
             cls.permissions.pop('TemplateInvocation')
+        if 'rubygem-foreman_puppet' not in rpm_packages:
```

Comment:
There are more rpm conditions than previously. Have tests failed on these in upstream?

Reply:
Since upstream is defined as 
```
# rpm -q satellite-capsule
package satellite-capsule is not installed
```
yes they failed.

Reply:
https://github.com/SatelliteQE/robottelo/blob/21580fc40d64e0daaa31bf72d135bdd4d1fc9469/robottelo/hosts.py#L1540-L1547

What I got from your comment is that the current implementation of `is_upstream` does not suit you. Is that correct?

Reply:
I do not really have an issue with it. It is just more a check for 'not downstream satellite'.
I am not really running the tests on upstream but a different downstream so it kind of helps me I guess.

Reply:
@dosas is correct here that the missing presence of Satellite rpms isn't the most accurate test of whether we're in an upstream/downstream environment or not.

Reply:
@lhellebr To answer your original questions: Yes there are more conditions because there are also more permissions that were added.

IMO the definition of `is_upstream` is out of scope of this PR.